src/Factory/Api/Mastodon/FollowRequest.php

```diff
@@ -49,18 +50,12 @@ public function __construct(LoggerInterface $logger, BaseURL $baseURL)
 	 */
 	public function createFromIntroduction(Introduction $introduction): \Friendica\Object\Api\Mastodon\FollowRequest
 	{
-		$cdata = Contact::getPublicAndUserContactID($introduction->cid, $introduction->uid);
-
-		if (empty($cdata)) {
+		$account = DBA::selectFirst('account-user-view', [], ['id' => $introduction->cid, 'uid' => [0, $introduction->uid]], ['order' => ['id' => true]]);
```

Comment:
Shouldn't it be more like:
```
$account = DBA::selectFirst('account-user-view', [], ['id' => $introduction->cid, 'uid' => $introduction->uid]);
if (empty($account)) {
	$account = DBA::selectFirst('account-user-view', [], ['pid' => $introduction->cid, 'uid' => $introduction->uid]);
}
```

Reply:
What if there's no user-specific contact? Wouldn't both `account-user-view` queries return nothing? I still stand by my code, and here are the two different cases:

User-specific contact exists
------
```
id => User-specific contact id
pid => Public contact id
uid => User id
```
User-specific contact doesn't exist
-----
```
id => Public contact id
pid => Public contact id
uid => 0
```

In both cases, we query the relevant id on `id` because it would represent either the user-specific contact id or the public contact id, and we have to vary the `uid` unlike in your suggestion where the public contact wouldn't be queried at all if a user-specific contact doesn't exist.

Reply:
The endpoint is for pending follow requests: https://docs.joinmastodon.org/methods/follow_requests/#get

At this time there should always be a user contact (with `pending` set to true). If this was not the case, then something went completely wrong here.

Reply:
Through the DFRN suggestion code path, there is no user-specific contact, but the created introduction can still be retrieved by this endpoint.

Reply:
I just had got a look at the table on my system. If the content of `contact-id` is passed without any changes, then the contact is always a user contact and not a public one.

Reply:
I don't believe the DFRN suggestion code path is ever used, but it still exists, so we should support it. Of course, if we remove it, it would make things more straightforward in this PR. 👀

Reply:
But nonetheless the ` ['order' => ['id' => true]]` is superfluous here, since there will be only a single entry with that `id`.

Reply:
Yes, you're right.